### PR TITLE
LibJS: Add Date methods: setHours, setMinutes, setSeconds, setMilliseconds

### PR DESCRIPTION
**Pull request by tuqqu**

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/CommonPropertyNames.h [hunk 1]
```diff
@@ -201,6 +201,10 @@ namespace JS {
     P(round)                                 \
     P(set)                                   \
     P(setFullYear)                           \
+    P(setHours)                              \
+    P(setMinutes)                            \
+    P(setSeconds)                            \
+    P(setMilliseconds)                       \
     P(setPrototypeOf)                        \
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/CommonPropertyNames.h:208]: Please keep this sorted :)

File: Userland/Libraries/LibJS/Runtime/DatePrototype.cpp [hunk 2]
```diff
@@ -144,6 +148,48 @@ JS_DEFINE_NATIVE_FUNCTION(DatePrototype::get_hours)
     return Value(static_cast<double>(this_object->hours()));
 }
 
+JS_DEFINE_NATIVE_FUNCTION(DatePrototype::set_hours)
+{
+    auto* this_object = typed_this(vm, global_object);
+    if (!this_object)
+        return {};
+
+    auto new_hours = vm.argument(0).to_i32(global_object);
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/DatePrototype.cpp:157]: this should use `to_number()` - `to_i32()` will hide NaN and Infinity, but we don't have invalid dates yet so fine for now I guess

File: Userland/Libraries/LibJS/Runtime/DatePrototype.cpp [hunk 3]
```diff
@@ -144,6 +148,48 @@ JS_DEFINE_NATIVE_FUNCTION(DatePrototype::get_hours)
...
+    if (vm.argument_count() >= 4) {
+        auto new_milliseconds = vm.argument(3).to_i32(global_object);
+        if (vm.exception())
+            return {};
+        new_seconds += new_milliseconds / 1000;
+        this_object->set_milliseconds(new_milliseconds % 1000);
+    }
+
+    datetime.set_time(datetime.year(), datetime.month(), datetime.day(), new_hours, new_minutes, new_seconds);
+    return Value { this_object->time() };
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/DatePrototype.cpp:190]: Please use `Value()` for consistency with 99.99% of LibJS (might as well fix the 0.01% of `Value {}` @awesomekling introduced recently :stuck_out_tongue:)

File: Userland/Libraries/LibJS/Runtime/DatePrototype.cpp [hunk 4]
```diff
@@ -152,6 +198,27 @@ JS_DEFINE_NATIVE_FUNCTION(DatePrototype::get_milliseconds)
...
+
+    this_object->set_milliseconds(new_milliseconds % 1000);
+
+    auto added_seconds = new_milliseconds / 1000;
+    if (added_seconds > 0) {
+        auto& datetime = this_object->datetime();
+        datetime.set_time(datetime.year(), datetime.month(), datetime.day(), datetime.hour(), datetime.minute(), datetime.second() + added_seconds);
+    }
+
+    return Value { this_object->time() };
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/DatePrototype.cpp:219]: same as above

File: Userland/Libraries/LibJS/Runtime/DatePrototype.cpp [hunk 5]
```diff
@@ -160,6 +227,39 @@ JS_DEFINE_NATIVE_FUNCTION(DatePrototype::get_minutes)
...
+    if (vm.argument_count() >= 3) {
+        auto new_milliseconds = vm.argument(2).to_i32(global_object);
+        if (vm.exception())
+            return {};
+        new_seconds += new_milliseconds / 1000;
+        this_object->set_milliseconds(new_milliseconds % 1000);
+    }
+
+    datetime.set_time(datetime.year(), datetime.month(), datetime.day(), datetime.hour(), new_minutes, new_seconds);
+    return Value { this_object->time() };
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/DatePrototype.cpp:260]: same as above

File: Userland/Libraries/LibJS/Runtime/DatePrototype.cpp [hunk 6]
```diff
@@ -176,6 +276,30 @@ JS_DEFINE_NATIVE_FUNCTION(DatePrototype::get_seconds)
...
+        if (vm.exception())
+            return {};
+        new_seconds += new_milliseconds / 1000;
+        this_object->set_milliseconds(new_milliseconds % 1000);
+    }
+
+    auto& datetime = this_object->datetime();
+
+    datetime.set_time(datetime.year(), datetime.month(), datetime.day(), datetime.hour(), datetime.minute(), new_seconds);
+    return Value { this_object->time() };
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/DatePrototype.cpp:300]: same as above

**Review by linusg — CHANGES_REQUESTED** [review]

**Comment by tuqqu** [comment]
@linusg thanks for the review! I made changes

**Comment by linusg** [comment]
Please squash the commits and preferably add some basic tests as well :)

**Review by linusg — APPROVED** [review]

**Pull request merged**